tests/osaka/eip7939_count_leading_zeros/test_count_leading_zeros.py

```diff
@@ -155,6 +156,56 @@ def test_clz_stack_underflow(state_test: StateTestFiller, pre: Alloc):
     state_test(pre=pre, post=post, tx=tx)
 
 
+@pytest.mark.valid_from("Osaka")
+def test_clz_stack_not_overflow(state_test: StateTestFiller, pre: Alloc, fork: Fork):
+    """Test CLZ opcode never causes stack overflow."""
+    max_stack_items = fork.max_stack_height()
+
+    code = Bytecode()
+    post = {}
+
+    code += Op.PUSH0 * (max_stack_items - 2)
+
+    for i in range(256):
+        code += Op.PUSH1(i) + Op.CLZ(1 << i) + Op.SWAP1 + Op.SSTORE
+
+    code_address = pre.deploy_contract(code=code)
+
+    post[code_address] = Account(storage={i: 255 - i for i in range(256)})
+
+    tx = Transaction(
+        to=code_address,
+        sender=pre.fund_eoa(),
+        gas_limit=6_000_000,
+    )
+
+    state_test(pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.parametrize("bit", list(range(1, 33)))
+@pytest.mark.valid_from("Osaka")
+def test_clz_push_operation_same_value(state_test: StateTestFiller, pre: Alloc, bit: int):
+    """Test CLZ opcode returns the same value via different push operations."""
+    code = Bytecode()
+
+    for i in range(bit, 32):
+        op = getattr(Op, f"PUSH{i}")
+        code += Op.SSTORE(op[i], Op.CLZ(1 << bit))
```

Comment:
I feel we should use `Op.CLZ(op[1 << bit])` to guarantee that the PUSHn opcode is the one that pushes to the stack right before CLZ.